src/modules/Search/Search.js

```diff
@@ -0,0 +1,698 @@
+
+    const { minCharacters } = this.props
+    const { value } = this.state
+    if (value.length < minCharacters) return
+
+    e.preventDefault()
+    // TODO open/close should only change state, open/closeMenu should be called on did update
+    this.trySetState({ open: true })
+  }
+
+  selectItemOnEnter = (e) => {
+    debug('selectItemOnEnter()')
+    debug(keyboardKey.getName(e))
+    if (keyboardKey.getCode(e) !== keyboardKey.Enter) return
+    e.preventDefault()
+
+    const result = this.getSelectedResult()
+
+    // prevent selecting null if there was no selected item value
+    if (!result) return
+
+    // notify the onChange prop that the user is trying to change value
+    this.setValue(result.title)
+    this.onChange(e, result)
+    this.close()
+  }
+
+  closeOnDocumentClick = (e) => {
+    debug('closeOnDocumentClick()')
+    debug(e)
+    this.close()
+  }
+
+  // ----------------------------------------
+  // Component Event Handlers
+  // ----------------------------------------
+
+  handleItemClick = (e, id) => {
+    debug('handleItemClick()')
+    debug(id)
+    const result = this.getSelectedResult(id)
+
+    // prevent toggle() in handleClick()
+    e.stopPropagation()
+
+    // notify the onChange prop that the user is trying to change value
+    this.setValue(result.title)
+    this.onChange(e, result)
+    this.close()
+  }
+
+  handleFocus = (e) => {
+    debug('handleFocus()')
+    const { onFocus } = this.props
+    if (onFocus) onFocus(e)
+    this.setState({ focus: true })
+    // TODO
+    // handleFocus() is called on mouse down
+    // handleClick() it called on mouse up
+    // open() here is immediately toggled to close() by click
+    //
+    // const { value, open } = this.state
+    // if (value && !open) this.open()
+  }
+
+  handleBlur = (e) => {
+    debug('handleBlur()')
+    const { onBlur } = this.props
+    if (onBlur) onBlur(e)
+    this.setState({ focus: false })
+  }
+
+  handleSearchChange = (e) => {
+    debug('handleSearchChange()')
+    debug(e.target.value)
+    // prevent propagating to this.props.onChange()
+    e.stopPropagation()
+    const { onSearchChange, minCharacters } = this.props
+    const { open } = this.state
+    const newQuery = e.target.value
+
+    if (onSearchChange) onSearchChange(e, newQuery)
+
+    // open search dropdown on search query
+    if (newQuery.length < minCharacters) {
+      this.close()
+    } else if (!open) {
+      this.open()
+    }
+
+    this.setValue(newQuery)
+  }
+
+  // ----------------------------------------
+  // Getters
+  // ----------------------------------------
+
+  getResults = () => {
+    const { category, results, source, searchFields, maxResults } = this.props
+    const { value } = this.state
+
+    // filter by search query
+    if (!source) return results
+
+    const re = new RegExp(_.escapeRegExp(value), 'i')
+
+    const isMatch = (result) =>
+      _.some(searchFields, (field) => result[field] && re.test(result[field]))
```

Comment:
Since the regex will correctly pass/fail against any field value, the extra `result[field]` is not necessary.